plugins/outputs/influxdb_v2/influxdb.go

```diff
@@ -0,0 +1,199 @@
+package influxdb_v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers/influx"
+)
+
+var (
+	defaultURL = "http://localhost:9999"
+
+	ErrMissingURL = errors.New("missing URL")
+)
+
+var sampleConfig = `
+  ## The URLs of the InfluxDB cluster nodes.
+  ##
+  ## Multiple URLs can be specified for a single cluster, only ONE of the
+  ## urls will be written to each interval.
+  urls = ["http://127.0.0.1:9999"]
+
+  ## Token for authentication.
+  token = ""
+
+  ## Organization is the name of the organization you wish to write to; must exist.
+  organization = ""
+
+  ## Bucket to the name fo the bucketwrite into; must exist.
+  bucket = ""
+
+  ## Precision for the unix timestamps within the body line-protocol.
+  # precision = "ns"
+
+  ## Timeout for HTTP messages.
+  # timeout = "5s"
+
+  ## Additional HTTP headers
+  # http_headers = {"X-Special-Header" = "Special-Value"}
+
+  ## HTTP Proxy override, if unset values the standard proxy environment
+  ## variables are consulted to determine which proxy, if any, should be used.
+  # http_proxy = "http://corporate.proxy:3128"
+
+  ## HTTP User-Agent
+  # user_agent = "telegraf"
+
+  ## Content-Encoding for write request body, can be set to "gzip" to
+  ## compress body or "identity" to apply no encoding.
+  # content_encoding = "identity"
+
+  ## Optional TLS Config for use on HTTP connections.
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+type Client interface {
+	Write(context.Context, []telegraf.Metric) error
+
+	URL() string // for logging
+}
+
+type InfluxDB struct {
+	URLs            []string          `toml:"urls"`
+	Token           string            `toml:"token"`
+	Organization    string            `toml:"organization"`
+	Bucket          string            `toml:"bucket"`
+	Precision       string            `toml:"precision"`
+	Timeout         internal.Duration `toml:"timeout"`
+	HTTPHeaders     map[string]string `toml:"http_headers"`
+	HTTPProxy       string            `toml:"http_proxy"`
+	UserAgent       string            `toml:"user_agent"`
+	ContentEncoding string            `toml:"content_encoding"`
+	tls.ClientConfig
+
+	clients    []Client
+	serializer *influx.Serializer
+}
+
+func (i *InfluxDB) Connect() error {
+	ctx := context.Background()
+
+	if len(i.URLs) == 0 {
+		i.URLs = append(i.URLs, defaultURL)
+	}
+
+	i.serializer = influx.NewSerializer()
+	i.serializer.SetFieldTypeSupport(influx.UintSupport)
```

Comment:
@danielnelson , yes.  We support uint.

Reply:
Is uint always desired?  If someone imported non-uint data into the database this would be a change of type and could result in data be uninsertable.

Reply:
Sounds like an issue for db migrations, right? This flag only allows the plugin to add uints if they are configured/used, if i understand correctly.

Reply:
Plus, with a major version change there are bound to have some incompatibilities. Keeping this always set seems like the best thing since the database supports uints from the getgo.

Reply:
I don't think a data migration could handle this since the database doesn't know what should be a int/uint.  When it comes to Telegraf generated data, the only way to determine it today is by reading the code.